app/src/main/java/it/niedermann/owncloud/notes/persistence/NoteServerSyncHelper.java

```diff
@@ -92,7 +92,7 @@ public void onReceive(Context context, Intent intent) {
     // current state of the synchronization
     private final Map<String, Boolean> syncActive = new HashMap<>();
     private final Map<String, Boolean> syncScheduled = new HashMap<>();
-    private final NotesClient notesClient;
+    private final Map<String, NotesClient> notesClients = new HashMap<>();
```

Comment:
I suggest to remove this "cache" for NotesClients. NotesClients are state-less (there is no object variable), so the only thing that is won with this cache is omitting a single object creation on each server request. However, since the constructor is very simple, we are not saving a significant amount of time.

The drawback of this cache is, that it makes the code rather complicated and you have to ensure, that you are not using an outdated client (i.e. with an old API version), which is error-prone.

Therefore, I strongly suggest to remove this, since it's drawbacks are dominating.

Reply:
Done